kernel/src/scan/log_replay.rs

```diff
@@ -348,24 +388,10 @@ pub(crate) fn scan_action_iter(
     transform: Option<Arc<Transform>>,
     physical_predicate: Option<(ExpressionRef, SchemaRef)>,
 ) -> impl Iterator<Item = DeltaResult<ScanData>> {
-    let mut log_scanner = LogReplayScanner::new(engine, physical_predicate);
-    let add_transform = engine.get_expression_handler().get_evaluator(
-        get_log_add_schema().clone(),
-        get_add_transform_expr(),
-        SCAN_ROW_DATATYPE.clone(),
-    );
-    action_iter
-        .map(move |action_res| {
-            let (batch, is_log_batch) = action_res?;
-            log_scanner.process_scan_batch(
-                add_transform.as_ref(),
-                batch.as_ref(),
-                logical_schema.clone(),
-                transform.clone(),
-                is_log_batch,
-            )
-        })
-        .filter(|res| res.as_ref().map_or(true, |(_, sv, _)| sv.contains(&true)))
+    let log_scanner =
+        ScanLogReplayProcessor::new(engine, physical_predicate, logical_schema, transform);
+
+    log_scanner.process_actions_iter(action_iter)
```

Comment:
nit:
```suggestion
    ScanLogReplayProcessor::new(engine, physical_predicate, logical_schema, transform)
        .process_actions_iter(action_iter)
```
alt: Instead of creating an instance at all via `new`, just define a static method that does it all?
```suggestion
    ScanLogReplayProcessor::process_actions(
        engine, 
        physical_predicate, 
        logical_schema, 
        transform, 
        action_iter)
```

Reply:
Nit makes sense 👍 , The alternative suggestion bring some optional construction logic into the trait, which could improve ergonomics but might also introduce some design questions around standardizing processor setup. I think this is worth tracking as a design decision: #807